mtags/src/main/scala-3/scala/meta/internal/pc/PcSemanticTokensProvider.scala

```diff
@@ -60,18 +60,25 @@ final class PcSemanticTokensProvider(
   object Collector extends PcCollector[Option[Node]](driver, params):
     override def collect(
         parent: Option[Tree]
-    )(tree: Tree, pos: SourcePosition, symbol: Option[Symbol]): Option[Node] =
-      val sym = symbol.fold(tree.symbol)(identity)
-      if !pos.exists || sym == null || sym == NoSymbol then None
-      else
-        Some(
-          makeNode(
-            sym = sym,
-            pos = adjust(pos)._1,
-            isDefinition = isDefinition(tree),
-            isDeclaration = isDeclaration(tree),
-          )
-        )
+    )(
+        tree: Tree | EndMarker,
+        pos: SourcePosition,
+        symbol: Option[Symbol],
+    ): Option[Node] =
+      tree match
+        case tree: Tree =>
+          val sym = symbol.fold(tree.symbol)(identity)
+          if !pos.exists || sym == null || sym == NoSymbol then None
+          else
+            Some(
+              makeNode(
+                sym = sym,
+                pos = adjust(pos)._1,
+                isDefinition = isDefinition(tree),
+                isDeclaration = isDeclaration(tree),
+              )
+            )
+        case _ => None
```

Comment:
Maybe we can keep the symbol in end marker and this way provide good semantic token for it?

Reply:
Sure. By "good" we mean that it should be the same as the definition it ends?

Reply:
exactly